vm/vmgs/vmgs/src/vmgs_impl.rs

```diff
@@ -348,30 +343,15 @@ impl Vmgs {
             )));
         }
 
-        if let Err(e) = storage.validate_transfer_size(max_trans_size_bytes) {
-            return Err(Error::Initialization(format!("{}", e,)));
-        }
-
         // Any power-of-2 sector size up to 4096 bytes works, but in practice only 512 and 4096
         // indicate a supported (tested) device configuration.
-        if (logical_sector_size != 512 && logical_sector_size != 4096)
-            || (physical_sector_size != 512 && physical_sector_size != 4096)
-            || (physical_sector_size < logical_sector_size)
-        {
+        if sector_size != 512 && sector_size != 4096 {
             return Err(Error::Initialization(format!(
-                "Invalid sector size: logical={}, physical={}",
-                logical_sector_size, physical_sector_size
+                "Invalid sector size {}",
+                sector_size
             )));
         }
 
-        // Capacity and max transfer size should be multiple of the physical sector size
```

Comment:
any particular reason these checks got dropped?

Reply:
They're useless. We don't care what the physical sector size is at this layer.

If we care about the checks at all (which I don't), we should probably put them in the VMGS GET disk layer.

Reply:
Ok.

I might add some brief justification / explanation as to why why these params are useless somewhere else (maybe in the saved-state impl?)

Reply:
I added them back, but in the vmgs_get_disk implementation (which is more appropriate) instead of here. This also made the max transfer size thing make sense (and now we use it).